pkg/controller/packetcapture/packetcapture_controller.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/tigera/operator/pkg/render/common/networkpolicy"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+const (
+	PacketCaptureControllerName = "packet-capture-controller"
+	ResourceName                = "packet-capture"
+)
+
+var log = logf.Log.WithName("controller_packet_capture")
+
+// Add creates a new PacketCapture Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+
+	if !opts.EnterpriseCRDExists {
+		// No need to start this controller
+		return nil
+	}
+	licenseAPIReady := &utils.ReadyFlag{}
+	tierWatchReady := &utils.ReadyFlag{}
+
+	r := newReconciler(mgr, opts, licenseAPIReady, tierWatchReady)
+
+	c, err := ctrlruntime.NewController(PacketCaptureControllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create packetcapture-controller: %w", err)
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Error(err, "Failed to establish a connection to k8s")
+		return err
+	}
+
+	go utils.WaitToAddLicenseKeyWatch(c, k8sClient, log, licenseAPIReady)
+	go utils.WaitToAddTierWatch(networkpolicy.TigeraComponentTierName, c, k8sClient, log, tierWatchReady)
+
+	go utils.WaitToAddNetworkPolicyWatches(c, k8sClient, log, []types.NamespacedName{
+		{Name: render.PacketCapturePolicyName, Namespace: render.PacketCaptureNamespace},
+	})
+
+	if err = c.WatchObject(&operatorv1.PacketCaptureAPI{}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("packetcapture-controller failed to watch resource: %w", err)
+	}
+
+	if err = utils.AddSecretsWatch(c, render.PacketCaptureServerCert, common.OperatorNamespace()); err != nil {
+		return fmt.Errorf("packetcapture-controller failed to watch the Secret resource: %v", err)
+	}
+
+	if err = imageset.AddImageSetWatch(c); err != nil {
+		return fmt.Errorf("packetcapture-controller failed to watch ImageSet: %w", err)
+	}
+
+	if err = utils.AddTigeraStatusWatch(c, ResourceName); err != nil {
+		return fmt.Errorf("packetcapture-controller failed to watch packetcapture TigeraStatus: %w", err)
+	}
+
+	log.V(5).Info("Controller created and Watches setup")
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, opts options.AddOptions, licenseAPIReady *utils.ReadyFlag, tierWatchReady *utils.ReadyFlag) reconcile.Reconciler {
+	r := &ReconcilePacketCapture{
+		client:              mgr.GetClient(),
+		scheme:              mgr.GetScheme(),
+		provider:            opts.DetectedProvider,
+		enterpriseCRDsExist: opts.EnterpriseCRDExists,
+		status:              status.New(mgr.GetClient(), ResourceName, opts.KubernetesVersion),
+		clusterDomain:       opts.ClusterDomain,
+		usePSP:              opts.UsePSP,
+		licenseAPIReady:     licenseAPIReady,
```

Comment:
Do we need the license watch in this controller? Looks like we never query the license.

Reply:
@caseydavenport  I will remove this. Just wondering if we need a license check for packet capture. Initially, it was part of [apiserver_controller.go](https://github.com/tigera/operator/blob/1565f374e78473c889726665b5c21ba62be88549/pkg/controller/apiserver/apiserver_controller.go#L408), and it didn't have any license check at that time.